src/DependencyInjection/Compiler/RegisterControllersPass.php

```diff
@@ -83,4 +86,18 @@ private function buildControllerDefinitionFromClass(string $class) : Definition
 
         return $definition;
     }
+
+
+    /**
+     * @param Definition $definition
+     * @return Definition
+     */
+    private function injectDependencies(Definition $definition) : Definition
```

Comment:
I edited it, but i don't know how to deal with those 2 points.

1) If i check if trait is there, i would must check if `DoctrineAwareTrait` or `ControllerAwareTrait` is in controller, don't i? I would even don't use any trait and just put setDoctrineRegistry into controller and it would works. I don't konw if it is good or bad...
2) I couldn't find out how to use injectation without `[new Reference('doctrine')]`.

Even i think, i use it bad way. I do almost the same thinks as you do in `RegisterControllersPass`. 
Can you help me with it?